plugins/s3_auth/aws_auth_v4.cc

```diff
@@ -192,8 +192,22 @@ trimWhiteSpaces(const char *in, size_t inLen, size_t &newLen)
     last--;
   }
 
-  newLen = last - first + 1;
-  return first;
+  std::stringstream result;
+  int consecutiveSpaces = 0;
+  while (first <= last) {
+    if (*first == ' ') {
```

Comment:
Not sure this will work. Correct me if I'm wrong, but to me it looks like you're replacing consecutive spaces with one space, and putting it back in the string. So `access_key=    foo` will become `access_key= foo`. But from what I understand, #7888 wants all space removed. Also, I believe `isspace` checks for all white-space characters, not just space.

Btw, if we're removing all white-spaces in the string, why have separate loops to remove the leading and trailing spaces, this could all be done in one go.

Reply:
You're right. I didn't read that issue carefully. I opened a new issue describing what I'm actually trying to fix https://github.com/apache/trafficserver/issues/8583

Reply:
Regarding removing leading and trailing whitespaces separately it was done this way before. What's the idea to remove leading and trailing spaces in one loop? Going from begging to the end how do you know that trailing spaces started? In the issue I linked how to create canonical headers and from what I understand you need to remove leading and trailing whitespaces but you need to compact ordinary spaces only. If we were removing only spaces or all whitespaces it could be done in one iteration.

Reply:
Ok, nvm the "in one go" part, I thought the value can't have spaces in it, but looks like the value itself may contain spaces in between. 

But I would still suggest you fix both #7888 and #8583 at the same time, you will only need to add a check for `=`, should be pretty easy to do.

Reply:
Also, you need to fix your code, the Ubuntu build is failing at `test_aws_auth_v4.cc:239`.

Reply:
@duke8253 I managed to fix tests and add some new. I can fix #7888 in another pr as it affects different code (and file) and doesn't seem to be related to the change I proposed in this pr.

Reply:
Looks mostly good, could you change your condition check on line 199 to use `isspace()`? Keep it consistent as the trimming part of the code, and it checks for all white space characters, not just spaces.

Reply:
Hmm, ok, missed that part.